pandas/core/indexes/base.py

```diff
@@ -5462,3 +5470,19 @@ def default_index(n):
     from pandas.core.indexes.range import RangeIndex
 
     return RangeIndex(0, n, name=None)
+
+
+def maybe_extract_name(name, obj, cls) -> Optional[Hashable]:
```

Comment:
see my comment above, can be more general so need to relocate i think as this is common between series/index)